### PR TITLE
Collections related to search

### PR DESCRIPTION
**Pull request by alansutherland**

### CODE REVIEW

File: components/search/RelatedChip.vue [hunk 1]
```diff
@@ -0,0 +1,69 @@
+<template>
+  <b-badge
+    v-if="title"
+    :to="linkGen(id)"
+    pill
+    variant="light"
+    :class="img ? 'img-chip mt-1 mr-2 font-weight-normal bg-white' : 'mt-1 mr-2 font-weight-normal bg-white'"
```
[the remark this answers is not in the document]
Comment (alansutherland) [reply components/search/RelatedChip.vue:7]: thanks I didn't realise you could define both class and :class - this is much better

File: components/search/RelatedChip.vue [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+<style lang="scss" scoped>
+  @import './assets/scss/variables.scss';
+  .badge-light {
+    color: $black;
+    font-size: $font-size-small;
+    border-radius: 1.125rem;
+    box-shadow: $boxshadow-light;
+    padding: 0.25rem 0.75rem;
+    margin-bottom: 2rem;
```
[the remark this answers is not in the document]
Comment (alansutherland) [reply components/search/RelatedChip.vue:54]: yep makes sense, have moved to generic

File: components/search/RelatedChip.vue [hunk 3]
```diff
@@ -0,0 +1,69 @@
+<template>
```
[the remark this answers is not in the document]
Comment (alansutherland) [reply components/search/RelatedChip.vue:1]: yep makes sense, have moved to generic

File: components/search/RelatedChip.vue (hunk 2 above)
[the remark this answers is not in the document]
Comment (alansutherland) [reply components/search/RelatedChip.vue:54]: I've reduced to 0.25rem

File: components/search/RelatedChip.vue [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+    &.img-chip {
+      padding: 0.25rem 0.75rem 0.25rem 0.25rem;
+    }
+
+    img {
+      width: 28px;
+      height: 28px;
+    }
+  }
+</style>
```
[the remark this answers is not in the document]
Comment (alansutherland) [reply components/search/RelatedChip.vue:69]: Carlos has advised us to copy the material style and transition to whitesmoke background on hover

File: pages/search/index.vue [hunk 5]
```diff
@@ -61,6 +87,41 @@
...
+        const languageParam = Array.from(new Set([this.$i18n.locale])).join(',');
+
+        const suggestions = await getEntitySuggestions(query, {
+          language: languageParam
+        }, {
+          recordValidation: this.enableSuggestionValidation
+        });
+
+        let i = 0;
+        while (i < 4) {
```
[the remark this answers is not in the document]
Comment (alansutherland) [reply pages/search/index.vue:106]: ah yes slice is a much neater solution, have updated

File: pages/search/index.vue [hunk 6]
```diff
@@ -61,6 +95,39 @@
...
+          recordValidation: this.enableSuggestionValidation
+        });
+
+        this.relatedCollections = suggestions.slice(0, 4);
+      },
+
+      suggestionLinkGen(entityUri) {
+        const entity = {
+          id: entityUri,
+          prefLabel: this.relatedCollections[entityUri]
```
Comment (mirjamverloop) [inline pages/search/index.vue:121]: As discussed, this seems to be always empty. Could you see if you can change that?

Comment (alansutherland) [reply pages/search/index.vue]: now fixed

File: components/generic/RelatedChip.vue [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+    name: 'RelatedChip',
+
+    props: {
+      linkGen: {
+        type: Function,
+        default: (val) => val
+      },
+      title: {
+        type: String,
+        default: () => {}
```
Comment (rwd) [inline components/generic/RelatedChip.vue:33]: If there is no title, then nothing is rendered. In other words, title is required. Make it so in the prop definition and remove the v-if on the b-badge.

File: components/generic/RelatedChip.vue [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+      },
+      id: {
+        type: String,
+        default: () => {}
```
Comment (rwd) [inline components/generic/RelatedChip.vue:37]: The default should be a string.

File: pages/search/index.vue [hunk 9]
```diff
@@ -61,6 +95,35 @@
...
+      if (this.$route.query.query.length > 0) {
+        this.getSearchSuggestions(this.$route.query.query);
+      }
+    },
+
+    methods: {
+      async getSearchSuggestions(query) {
+
+        // Query in the user's language, removing duplicates
+        const languageParam = Array.from(new Set([this.$i18n.locale])).join(',');
```
Comment (rwd) [inline pages/search/index.vue:107]: This is redundant. Just set below: `language: this.$i18n.locale`

File: pages/search/index.vue [hunk 10]
```diff
@@ -61,6 +95,35 @@
...
+
+        const suggestions = await getEntitySuggestions(query, {
+          language: languageParam
+        }, {
+          recordValidation: this.enableSuggestionValidation
```
Comment (rwd) [inline pages/search/index.vue:112]: Remove this as the record validation option is to be removed in another open PR.

File: components/generic/RelatedCollections.vue [hunk 11]
```diff
@@ -0,0 +1,83 @@
...
+      this.getSearchSuggestions(this.query);
+    },
+
+    mounted() {
+      this.$fetch();
+    },
+
+    methods: {
+      async getSearchSuggestions(query) {
+        const suggestions = await getEntitySuggestions(query, {
```
Comment (rwd) [inline components/generic/RelatedCollections.vue:66]: Suggestions are showing even if the query is empty, but should not.

Resolve with the logic in the diff in the following comment.

File: components/generic/RelatedCollections.vue [hunk 12]
```diff
@@ -0,0 +1,83 @@
...
+      query() {
+        this.$fetch();
+      }
+    },
+
+    fetch() {
+      this.getSearchSuggestions(this.query);
+    },
+
+    mounted() {
```
Comment (rwd) [inline components/generic/RelatedCollections.vue:60]: This is redundant and should be removed.

File: components/generic/RelatedCollections.vue [hunk 13]
```diff
@@ -0,0 +1,83 @@
...
+    },
+
+    computed: {
+      ...mapGetters({
+        apiConfig: 'apis/config'
+      })
+    },
+
+    watch: {
+      query() {
```
Comment (rwd) [inline components/generic/RelatedCollections.vue:51]: This can be reduced to `query: '$fetch'`

File: components/generic/RelatedCollections.vue [hunk 14]
```diff
@@ -0,0 +1,83 @@
...
+    mounted() {
+      this.$fetch();
+    },
+
+    methods: {
+      async getSearchSuggestions(query) {
+        const suggestions = await getEntitySuggestions(query, {
+          language: this.$i18n.locale
+        });
+        this.relatedCollections = suggestions.slice(0, 4);
```
Comment (rwd) [inline components/generic/RelatedCollections.vue:69]: This results in 10 suggestions coming back from the API. Let's limit the number requested, with:
```diff
diff --git a/components/generic/RelatedCollections.vue b/components/generic/RelatedCollections.vue
index 15a142fd..95d0f696 100644
--- a/components/generic/RelatedCollections.vue
+++ b/components/generic/RelatedCollections.vue
@@ -63,10 +63,10 @@
 
     methods: {
       async getSearchSuggestions(query) {
-        const suggestions = await getEntitySuggestions(query, {
-          language: this.$i18n.locale
+        this.relatedCollections = query === '' ? [] : await getEntitySuggestions(query, {
+          language: this.$i18n.locale,
+          rows: 4
         });
-        this.relatedCollections = suggestions.slice(0, 4);
       },
 
       suggestionLinkGen(id, prefLabel) {
diff --git a/plugins/europeana/entity.js b/plugins/europeana/entity.js
index cb9ddffb..c9cf311f 100644
--- a/plugins/europeana/entity.js
+++ b/plugins/europeana/entity.js
@@ -33,14 +33,13 @@ function entityApiUrl(endpoint) {
  * Get entity suggestions from the API
  * @param {string} text the query text to supply suggestions for
  * @param {Object} params additional parameters sent to the API
- * @param {string} params.language language(s), comma-separated, to request
  */
 export function getEntitySuggestions(text, params = {}) {
   return axios.get(entityApiUrl('/suggest'), {
     params: {
+      ...params,
       text,
       type: 'agent,concept',
-      language: params.language,
       scope: 'europeana',
       wskey: config.entity.key
     }
```

**Pull request merged**